modules/local/presto/presto_maskprimers.nf

```diff
@@ -51,3 +51,67 @@ process PRESTO_MASKPRIMERS {
     }
 
 }
+
+process PRESTO_MASKPRIMERS_POSTASSEMBLY {
+    tag "$meta.id"
+    label "process_high"
+
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::presto=0.6.2=py_0" : null)              // Conda package
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/presto:0.6.2--py_0"  // Singularity image
+    } else {
+        container "quay.io/biocontainers/presto:0.6.2--py_0"                        // Docker image
+    }
+
+    input:
+    tuple val(meta), path(R1)
+    path(cprimers)
+    path(vprimers)
+
+    output:
+    tuple val(meta), path("*REV_primers-pass.fastq") , emit: reads
+    path "*command_log.txt", emit: logs
+    path "*.log"
+    path "*.tab", emit: log_tab
+
+
+    script:
+    def primer_start_R1 = "--start ${params.cprimer_start}"
+    def primer_start_R2 = "--start ${params.vprimer_start}"
+    def revpr = ''
+    if (params.umi) {
+        primer_start_R1 = (params.index_file | params.umi_position == 'R1') ? "--start ${params.umi_length + params.cprimer_start} --barcode" : "--start ${params.cprimer_start}"
+        primer_start_R2 = (params.umi_position == 'R2') ? "--start ${params.umi_length + params.vprimer_start} --barcode" : "--start ${params.vprimer_start}"
+    }
+    if (params.maskprimers_revpr) {
+        revpr = '--revpr'
+    }
+    if (params.cprimer_position == "R1") {
+        """
+        MaskPrimers.py score --nproc ${task.cpus} -s $R1 -p ${cprimers} $primer_start_R1 --maxerror ${params.primer_maxerror} \
+            --mode ${params.primer_mask_mode} --outname ${meta.id}-FWD \
```

Comment:
I was following the [convention from the example workflow](https://presto.readthedocs.io/en/stable/workflows/Greiff2014_Workflow.html#read-annotation-and-masking-of-primer-regions)- I suppose we could change it to R1/R2 but since we have already joined the pairs FWD/REV makes intuitive sense to me. But I'm happy to go with your preference on this. I think now that this is in its own postassembly modulefile it should confuse people working with the more standard with-UMI option less :-)